api/src/main/java/com/marketplace/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+package com.marketplace.service.impl;
+
+import com.marketplace.dto.ProductDto;
+import com.marketplace.entity.Product;
+import com.marketplace.exception.NotFoundException;
+import com.marketplace.exception.UnavailableProductException;
+import com.marketplace.exception.IllegalArgumentException;
+import com.marketplace.repository.ProductRepository;
+import com.marketplace.service.ProductService;
+import com.marketplace.utils.mapper.ProductMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Implementation of the ProductService interface, providing CRUD operations
+ * for product management.
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ProductServiceImpl implements ProductService {
+
+    private final ProductRepository productRepository;
+
+    /**
+     * Creates a new product entity based on the provided DTO.
+     *
+     * @param productDto The DTO containing the product details.
+     * @return The saved product entity.
+     */
+    @Override
+    public ProductDto createProduct(ProductDto productDto) {
+        log.info("Received ProductDto: {}", productDto);
+        if (productDto.getMinQuantity() >= productDto.getMaxQuantity()) {
+            throw new IllegalArgumentException("La quantité minimale doit être inférieure à la quantité maximale");
+        }
+        Product product = ProductMapper.INSTANCE.toEntity(productDto);
+        log.info("Converted Product: {}", product);
+        Product savedProduct = productRepository.save(product);
+        return ProductMapper.INSTANCE.toDto(savedProduct);
+    }
+
+    /**
+     * Retrieves all products with stock > 0 for customers.
+     *
+     * @return A list of available products.
+     */
+    @Override
+    public List<Product> getAvailableProducts() {
+        log.info("Retrieving products with stock > 0 and active status");
+
+        List<Product> products = productRepository.findByActiveAndStockQuantityGreaterThan(true, 0);
+
+        if (products.isEmpty()) {
+            log.warn("No products found with stock > 0 found and active status.");
+            return Collections.emptyList();
+        }
+            return products;
+    }
+
+
+    /**
+     * Retrieves a product entity by its ID.
+     *
+     * @param id The ID of the product to retrieve.
+     * @return An Optional containing the product entity if found.
+     */
+    @Override
+    public Product getProductById(Long id) {
+        return productRepository.findById(id)

```

Comment:
Pour gagner en performance et séparer correctement les responsabilités des class, il est préférable de déplacer toute la logique dans ton repository. Par ex : 
@Query("SELECT p FROM Product p WHERE p.id = :id AND p.active = true")
Optional<Product> findByIdAndActive(@Param("id") Long id);

Comme ça dans ton service t'as juste à appeler cette requête et faire un return